ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -83,6 +83,14 @@ public UIManagerModule(
     reactContext.addLifecycleEventListener(this);
   }
 
+  /**
+   * This method gives an access to the underlying native view hierarchy manager. Note that all the
+   * operations executed on that object should be run in the UI thread. Use with caution
+   */
+  public NativeViewHierarchyManager getNativeViewHierarchyManager() {
+    return mUIImplementation.getUIViewOperationQueue().getNativeViewHierarchyManager();
```

Comment:
Not really a fan of exposing these objects. I think it might make more sense to integrate this all directly into UIManagerModule: there is definitely already implicit coupling (e.g. the ordering of callbacks in ReactChoreographer)